### ISSUE TITLE
Mixer View Ticks are not updated correctly across insistences of the app

### ISSUE THREAD

**Issue by zacjansheski** [musescore/MuseScore#17700]
### Issue type

UI bug

### Bug description

The "ticks" or "check marks" next to channels in the view options for the mixer are inconsistent when working with multiple instances of MuseScore.

### Steps to reproduce

1. Open two instances of MuseScore
2. Go to the mixer menu ...
3. Select view and tick or untick a channel.
4. The other instance of MuseScore will update
5. Check the mixer menu ... in the second window
6. Look at the check marks under view and they will be inconsistent

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/69917893/c5391c29-df72-425e-959a-ac7765f12a79

### MuseScore Version

MuseScore version (64-bit): 4.1.0-231420826, revision: github-musescore-musescore-230479a

### Regression

No.

### Operating system

OS: macOS 11.6, Arch.: x86_64, 

### Additional context

This is pertaining to the release of the mixer with added reverb controls.

https://github.com/musescore/MuseScore/pull/17570

**Comment by nasehim7** [comment]
Hi @zacjansheski @Eism  I am interested in taking this one up

**Comment by nasehim7** [comment]
Hi @zacjansheski 

I've got an understanding of how the Mixer Panel is set up in MuseScore. I saw that when clicking on the Aux Send and Aux Channel menu items, it updates the shared settings value, unlike other items that update the local value, i.e., properties in current instance. The mixer panel's context menu has methods that toggle the menu check status for its items locally. However, it doesn’t align well with send and channel menu items as their value is shared across instances, resulting in this inconsistent UI.

The approach that I was thinking about was to do the following:

1. We can have signals `auxSendVisibilityChanged` and `auxChannelVisibilityChanged` in the Mixer Panel context menu to indicate changes in the visibility of Aux Send and Aux Channel items, respectively, pushing out `index` and `newVisibilityValue.`
2. We can update In the constructor (`MixerPanelContextMenuModel::MixerPanelContextMenuModel`), establish connections between these signals and corresponding slots (`handleAuxSendVisibilityChange` and `handleAuxChannelVisibilityChange`) to handle visibility changes.
3. We will need to update the toggle methods (`toggleAuxSend` and `toggleAuxChannel`) to emit the signals (`auxSendVisibilityChanged` and `auxChannelVisibilityChanged`, respectively) instead of directly setting menu items checked status.
4. Lastly, create slots (`handleAuxSendVisibilityChange` and `handleAuxChannelVisibilityChange`) which will set menu items' check status based on received signals in all instances of the Mixer Panel context menu.

Let me know if my understanding/approach is in the right direction or am I missing something? :)

**Comment by cbjeukendrup** [comment]
@nasehim7 That's close, but what's missing in your analysis is PlaybackConfiguration (accessed via `configuration()` in MixerPanelContextMenuModel). 

In PlaybackConfiguration, the actual values are stored, and PlaybackConfiguration handles synchronisation of the values between instances (well, that happens yet one level deeper, namely in `Settings`, but that's a detail).
So PlaybackConfiguration is the ultimate source of truth, but should also be the ultimate source of notifications.

Let's consider the Aux Sends. They are toggled by `toggleAuxSend`. This function calls `setViewMenuItemChecked` and `emit auxSendsSectionVisibleChanged();`. This is not good: now `toggleAuxSend` is the ultimate source of these notifications! It's bad because `toggleAuxSend` is only one of the reasons that the checkmark state of the action may need to be reevaluated. As said, the thing that does know all the reasons is `PlaybackConfiguration`.

So, first we need to remove the calls to `setViewMenuItemChecked` and `emit auxSendsSectionVisibleChanged();` from `toggleAuxSend`. Instead, these calls need to happen whenever PlaybackConfiguration emits its notification. Specifically, in `MixerPanelContextMenuModel::load()`, we need to start listening for the `isAuxSendVisibleChanged()` notification from PlaybackConfiguration:
```c++
void MixerPanelContextMenuModel::load() 
{
    … // where exactly in `load()` you place it doesn't really matter

    configuration()->isAuxSendVisibleChanged().onReceive(this, [this](aux_channel_idx_t auxSendIndex, bool newVisibilityValue) {
        setViewMenuItemChecked(auxSendVisibleMenuItemId(auxSendIndex), newVisibilityValue);

        emit auxSendsSectionVisibleChanged();
    });

    …
}
```

To summarise, now the flow is always like this:
1. Either `toggleAuxSend` calls `configuration()->setAuxChannelVisible(…)`, or that happens in some other instance.
(`toggleAuxSend` does _not_ directly update the checkmark state of the action.)
2. This triggers the `configuration()->isAuxSendVisibleChanged()` notification to be emitted.
3. Since we (MixerPanelContextMenuModel) had registered a callback for this notification, this callback will be called now
4. Thus, the checkmark state of the action is updated, and the MixerPanelContextMenuModel emits `auxSendsSectionVisibleChanged()` to tell the Mixer UI to update too

**Comment by nasehim7** [comment]
@cbjeukendrup Sweet! Thanks a lot :D now I understand it. In the codebase, all the configuration files are the ones that usually handle all the notifications related to that component. Correct?

**Comment by cbjeukendrup** [comment]
Correct! Usually the …Configurations are the single source of truth, and the root of the notification tree, to use some buzzwords :)
